src/navigation/path.cpp

```diff
@@ -0,0 +1,472 @@
+#include <openspace/navigation/pathcurves/avoidcollisioncurve.h>
+#include <openspace/navigation/pathcurves/zoomoutoverviewcurve.h>
+#include <openspace/navigation/pathnavigator.h>
+#include <openspace/scene/scenegraphnode.h>
+#include <openspace/query/query.h>
+#include <openspace/util/collisionhelper.h>
+#include <openspace/util/universalhelpers.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/interpolator.h>
+
+namespace {
+    constexpr const char _loggerCat[] = "Path";
+    constexpr const float Epsilon = 1e-5f;
+
+    constexpr const char SunIdentifier[] = "Sun";
+
+    // TODO: where should this documentation be?
+    // It's nice to have these to interpret the dictionary when creating the path, but
+    // maybe it's not really necessary
+    struct [[codegen::Dictionary(PathInstruction)]] Parameters {
+        enum class Type {
+            Node,
+            NavigationState
+        };
+        Type type;
+
+        // The desired duration traversing the specified path segment should take
+        std::optional<float> duration;
+
+        // (Node): The target node of the camera path. Not optional for 'Node' instructions
+        std::optional<std::string> target;
+
+        // (Node): An optional position in relation to the target node, in model
+        // coordinates (meters)
+        std::optional<glm::dvec3> position;
+
+        // (Node): An optional height in relation to the target node, in meters
+        std::optional<double> height;
+
+        // (Node): If true, the up direction of the node is taken into account when
+        // computing the wayopoint for this instruction
+        std::optional<bool> useTargetUpDirection;
+
+        // (NavigationState): A navigation state that will be the target
+        // of this path segment
+        std::optional<ghoul::Dictionary> navigationState
+            [[codegen::reference("core_navigation_state")]];
+
+        // A navigation state that determines the start state for the camera path
+        std::optional<ghoul::Dictionary> startState
+            [[codegen::reference("core_navigation_state")]];
+    };
+#include "path_codegen.cpp"
+} // namespace
+
+namespace openspace::interaction {
+
+Path::Path(Waypoint start, Waypoint end, CurveType type,
+           std::optional<double> duration)
+    : _start(start), _end(end), _curveType(type)
+{
+    switch (_curveType) {
+        case CurveType::AvoidCollision:
+            _curve = std::make_unique<AvoidCollisionCurve>(_start, _end);
+            break;
+        case CurveType::Linear:
+            _curve = std::make_unique<LinearCurve>(_start, _end);
+            break;
+        case CurveType::ZoomOutOverview:
+            _curve = std::make_unique<ZoomOutOverviewCurve>(_start, _end);
+            break;
+        default:
+            LERROR("Could not create curve. Type does not exist!");
+            throw ghoul::MissingCaseException();
+    }
+
+    const auto defaultDuration = [](double pathlength) {
```

Comment:
If this is only used once on line 113, maybe it's better to just used the `log` function directly rather than going through a lambda?